dotCMS/src/main/java/com/dotcms/rest/api/v1/experiment/ExperimentFactory.java

```diff
@@ -0,0 +1,233 @@
+                ExperimentStatus.valueOf(experimentMap.get("status").toString()),
+               Boolean.parseBoolean(
+                       experimentMap.get("uniquePerVisitor").toString()),
+               Integer.parseInt(
+                       experimentMap.get("lookBackWindowMinutes").toString())
+       );
+
+        final List<Rule> rules = getRuleByExperiment(experiment);
+        final VariantsCollection variantsCollection = getVariantByExperiment(experiment,  experimentMap.get("pageId").toString());
+        final List<AnalyticEvent> events = getEventsByExperiment(experiment);
+
+        experiment.setEvents(events);
+        experiment.setRules(rules);
+        experiment.setVariantsCollection(variantsCollection);
+
+        return experiment;
+    }
+
+    private static List<AnalyticEvent> getEventsByExperiment(final Experiment experiment) {
+        try{
+            return new DotConnect()
+                    .setSQL(GET_EVENTS_BY_EXPERIMENT_QUERY)
+                    .addParam(experiment.getKey())
+                    .loadObjectResults()
+                    .stream()
+                    .map(map -> {
+                        try {
+                            return new AnalyticEvent(
+                                        AnalyticEventType.valueOf(map.get("event_key").toString()),
+                                        JsonUtil.toMap(map.get("parameters").toString())
+                                    );
+                        } catch (IOException e) {
+                            return null;
+                        }
+                    })
+                    .collect(Collectors.toList());
+        } catch (DotDataException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static List<Rule> getRuleByExperiment(final Experiment experiment) {
+        try {
+            return new DotConnect()
+                    .setSQL(GET_RULES_BY_EXPERIMENT_QUERY)
+                    .addParam(experiment.getKey())
+                    .loadObjectResults()
+                    .stream()
+                    .map(result -> result.get("rule_id").toString())
+                    .map(ruleID -> {
+                        try {
+                            return APILocator.getRulesAPI()
+                                    .getRuleById(ruleID, APILocator.systemUser(), false);
+                        } catch (DotDataException | DotSecurityException e) {
+                            return null;
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
+        } catch (DotDataException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static VariantsCollection getVariantByExperiment(final Experiment experiment,
+            final String pageInode) {
+        try {
+            final IHTMLPage page = APILocator.getHTMLPageAssetAPI()
```

Comment:
Same thing should not use API, should receive a IHtmlPage as an argument instead of using the API here